src/metabase/api/public.clj

```diff
@@ -233,6 +236,39 @@
      :export-format :api
      :parameters    parameters)))
 
+(api/defendpoint GET "/dashboard/:uuid/dashcard/:dashcard-id/execute/:slug"
+  "Fetches the values for filling in execution parameters. Pass PK parameters and values to select."
+  [uuid dashcard-id slug parameters]
+  {dashcard-id su/IntGreaterThanZero
+   slug su/NonBlankString
+   parameters su/JSONString}
+  (validation/check-public-sharing-enabled)
+  (let [dashboard-id (api/check-404 (db/select-one-id Dashboard :public_uuid uuid, :archived false))]
+    (actions.execution/fetch-values dashboard-id dashcard-id slug (json/parse-string parameters))))
+
+(def ^:private dashcard-execution-throttle (throttle/make-throttler :dashcard-id :attempts-threshold 5000))
+
+(api/defendpoint POST "/dashboard/:uuid/dashcard/:dashcard-id/execute/:slug"
+  "Execute the associated Action in the context of a `Dashboard` and `DashboardCard` that includes it.
+
+   `parameters` should be the mapped dashboard parameters with values.
+   `extra_parameters` should be the extra, user entered parameter values."
+  [uuid dashcard-id slug :as {{:keys [parameters], :as _body} :body}]
+  {dashcard-id su/IntGreaterThanZero
+   slug su/NonBlankString
+   parameters (s/maybe {s/Keyword s/Any})}
+  (let [throttle-info (try
+                        (throttle/check dashcard-execution-throttle dashcard-id)
+                        nil
+                        (catch ExceptionInfo e (ex-data e)))]
+    (if throttle-info
+      {:status 400 :body (get-in throttle-info [:errors :dashcard-id])}
```

Comment:
Wouldn't 429 status be more appropriate? Could we suggest a `Retry-After` value?